src/lib.rs

```diff
@@ -337,6 +338,12 @@ impl Parser {
                     // Continue collecting bytes into param
                     self.param *= 10;
                     self.param += (byte - b'0') as i64;
+
+                    // Mimic XTerm's overflow-prevention behavior
+                    if self.param > u16::MAX as i64 {
```

Comment:
It's kinda left up to the implementation as it turns out. For example, _tmux_ uses `INT_MAX` which should be equivalent to `i32::MAX`. I've tried to find the spec max value in ECMA48, but it's not mentioned.